src/org/omegat/core/data/PluginInformation.java

```diff
@@ -42,56 +50,91 @@
     private static final String BUNDLE_NAME = "Bundle-Name";
     private static final String BUILT_BY = "Built-By";
 
+    public enum Status {
+        INSTALLED,
+        BUNDLED,
+        UPGRADABLE,
+    }
+
     private final String className;
     private final String name;
     private final String version;
     private final String author;
     private final String description;
+    private final String category;
+    private final String link;
+    private final URL url;
+    private Status status;
 
-    public PluginInformation(String className, Manifest manifest) {
+    public PluginInformation(final String className, final Manifest manifest, final URL mu, final Status status) {
         this.className = className;
-        Attributes attrs = manifest.getMainAttributes();
-        name = findName(manifest);
-        version = findVersion(manifest);
-        author = findAuthor(manifest);
+        Attributes mainAttrs = manifest.getMainAttributes();
+        Attributes attrs = manifest.getEntries().get(className);
+        if (attrs == null) {
+              attrs = manifest.getMainAttributes();
+        }
+        name = findName(attrs);
+        version = findVersion(attrs, mainAttrs);
+        author = findAuthor(mainAttrs);
```

Comment:
I'm confused about the difference between `mainAttrs` and `attrs`.

If `attrs` is null (when would that be?) then we use `mainAttrs` for `attrs`.

Some information is looked up in `attrs` but some only in `mainAttrs`.

Why are some things in one but not the other? Can we not make the lookup logic simpler, e.g. always look in one, and then the other?

Also would it not be better to keep `PluginInformation` a POJO (keep this lookup logic out of the constructor)? There could be a factory method e.g. `buildFromManifest`.

Reply:
>  the difference between `mainAttrs` and `attrs`.

There is because `PluginInformation` is created from old API manifest,  and  new API manifest.

The old manifest looks like

```
Manifest-Version: 1.0
Bundle-ManifestVersion: 2
Bundle-Name: Okapi Filters for OmegaT
OmegaT-Plugin: true

Name: net.sf.okapi.lib.omegat.TSFilter
OmegaT-Plugin: filter

Name: net.sf.okapi.lib.omegat.TTXFilter
OmegaT-Plugin: filter
```

Here `Bundle Name` field is in `mainAttr`, and `Name: <class>` field is in `attr`.
When loading plugin like okapi filter, several `PluginInformation` objects are created with same 'mainAttr` and each `attr` 


When new one

```
Manifest-Version: 1.0
Plugin-Name: Okapi Filters for OmegaT
OmegaT-Plugins: net.sf.okapi.lib.omegat.OmegatOkapiFilter
```

All field is in `mainAttr` (or `attr`?) and another is null.

Reply:
> Also would it not be better to keep PluginInformation a POJO (keep this lookup logic out of the constructor)? There could be a factory method e.g. buildFromManifest.

I'd like to try it.

Reply:
I've introduced a `PluginInformationFactory` class to make things simple.